cmd/create/cluster/cmd.go

```diff
@@ -2571,20 +2571,23 @@ func run(cmd *cobra.Command, _ []string) {
 		r.Reporter.Errorf("There was a problem checking version compatibility: %v", err)
 		os.Exit(1)
 	}
+
 	additionalComputeSecurityGroupIds := args.additionalComputeSecurityGroupIds
 	getSecurityGroups(r, cmd, isVersionCompatibleComputeSgIds,
-		securitygroups.ComputeKind, useExistingVPC, isHostedCP, subnets,
+		securitygroups.ComputeKind, useExistingVPC, subnets,
 		subnetIDs, &additionalComputeSecurityGroupIds)
 
 	additionalInfraSecurityGroupIds := args.additionalInfraSecurityGroupIds
-	getSecurityGroups(r, cmd, isVersionCompatibleComputeSgIds,
-		securitygroups.InfraKind, useExistingVPC, isHostedCP, subnets,
-		subnetIDs, &additionalInfraSecurityGroupIds)
-
 	additionalControlPlaneSecurityGroupIds := args.additionalControlPlaneSecurityGroupIds
-	getSecurityGroups(r, cmd, isVersionCompatibleComputeSgIds,
-		securitygroups.ControlPlaneKind, useExistingVPC, isHostedCP, subnets,
-		subnetIDs, &additionalControlPlaneSecurityGroupIds)
+	if !isHostedCP {
+		getSecurityGroups(r, cmd, isVersionCompatibleComputeSgIds,
```

Comment:
I think you may now need to check if the flag changed for these two sets and return an error to the customer if it is hostedCp

Reply:
good catch, will add that along with any other requested changes soon